Add full-screen code functionality to and remove custom theming from EuiCodeBlock

cjcenizal opened the pull request:
This change also removes the `.euiCodeBlock--dark` and `.euiCodeBlock--light` classes. If you have CSS selectors which depend upon these classes then your `<EuiCodeBlock>` instances may appear broken. To reinstate the styles applied by those selectors, you'll have to apply those color values as SCSS color variables.

cjcenizal commented:
Good idea @snide! I'll add that.

cjcenizal commented:
@snide I made a few changes. Could you take a look?

* Added full-screen functionality to all code blocks.
* I changed the way highlighting works so that it adheres to the active theme. I think this makes the most sense in terms of how highlighting colors and theme colors interact.
* Made a few small adjustments to the way the `EuiImage` code is formatted.

snide commented on src/components/code/_code_block.scss:
```diff
@@ -14,6 +17,21 @@
+    left: 0;
+    right: 0;
+    bottom: 0;
+    padding: $euiSizeL !important;
```
I'd bump this up a notch since you have so much space to work with.

snide commented on src/global_styling/variables/_colors.scss:
```diff
@@ -54,3 +54,32 @@ $euiTextColor: $euiColorDarkestShade !default;
+
+// Code
+
+$euiCodeBlockBackgroundColor: #F5F5F5 !default;
```
This is just `$euiColorLightestShade`

snide commented on src/global_styling/variables/_colors.scss:
```diff
@@ -54,3 +54,32 @@ $euiTextColor: $euiColorDarkestShade !default;
+// Code
+
+$euiCodeBlockBackgroundColor: #F5F5F5 !default;
+$euiCodeBlockColor: #3F3F3F !default;
```
This is just `$euiTextColor`. Know these seems small, but saves people from needed to edit multiple vars.

cjcenizal replied:
Thanks good catch.

snide commented on src/components/image/image.js:
```diff
@@ -126,18 +128,18 @@ export class EuiImage extends Component {
       <figure
         className={classes}
-        onClick={this.toggleImageFullScreen}
+        onClick={allowFullScreen ? this.openFullScreen : undefined}
```
TY. This is a better way to do this.

snide commented on CHANGELOG.md:
```diff
@@ -1,6 +1,8 @@
-No public interface changes since `0.0.11`.
+**Breaking changes**
+
+- Removed `color` prop from `<EuiCodeBlock>`. This component's highlighting now matches whichever theme is currently active. [(#259)](https://github.com/elastic/eui/pull/259)
```
When we do a breaking change like this, can we make it a policy to put upgrade instructions in the PR itself. I think I did that for the last couple, and if not I should be! This ones funny though in that it won't technically break, it should just stylistically change.

cjcenizal replied:
I totally agree. Should I just add a note that this may result in a stylistic shift but no change in functionality, since `color` is now just a no-op? I figure that people will know the remove the use of this prop now that it no longer exists, but I can mention that too if you think we need to be that explicit.

snide replied:
Yeah. I think we just check on each other when we make merges to add something like the below...

## Upgrade path

This PR only changes the look of previous code blocks and does not require any changes if you are OK with the new styling. It will however break styling if you were applying custom code block themes beyond the default values. To retain that custom coloring, you'll want to apply those color values to the `colors.scss` files directly now.

cjcenizal replied:
Ahhh I think I get it! You mean we need to update the `CHANGELOG` with fallout caused by breaking changes to the _SCSS_, too right? I completely overlooked that, so thanks for pointing it out. I just pushed an update, let me know if I need to adjust it.

cjcenizal commented:
@snide Thanks for the solid feedback. I addressed your notes, could you take another look? I also added Nico as a reviewer, thanks for that suggestion. Your style improvements sound great. This PR was pretty quick-and-dirty from a design standpoint.

cjcenizal commented:
@uboness You can rebase `master` to get the full screen code example functionality now.
